rename_imports.py

```diff
@@ -0,0 +1,78 @@
+
+from LSP.plugin.core.file_watcher import FileWatcherEvent, FileWatcherProtocol, get_file_watcher_implementation, FileWatcher
+from LSP.plugin.core.typing import  List
+import os
+import sublime
+
+
+def register_rename_import_file_watcher(root_path: str):
+    file_watcher = get_file_watcher_implementation()
+    if file_watcher:
+        watcher = file_watcher.create(
+            root_path=root_path,
+            patterns=["**/*.ts", "**/*.tsx", "**/*.js", "**/*.jsx"],
+            events=["create", "delete"],
+            ignores=['**/node_modules/**'],
+            handler=HandleRenameImport
+        )
+
+
+queued_events = []  # type: List[FileWatcherEvent]
+
+
+class HandleRenameImport(FileWatcherProtocol):
+    def on_file_event_async(events: List[FileWatcherEvent]) -> None:
+        global queued_events
+        queued_events.append(events[0])
+        sublime.set_timeout(check_if_there_was_a_rename, 10)
```

Comment:
Doesn't seem like a reliable approach to have a timeout and check events from a different thread. This can result in race conditions and missing the two events you are looking for if there are more events besides that.

Reply:
I change it to set_timeout_async :)